### PR TITLE
MAT-5610 Reformat the way valuesets and codes are specified to get QDM CQL to work work in MADiE

### PR DESCRIPTION
**Pull request by adongare**
## Terminology Service PR

Jira Ticket: [MAT-5610](https://jira.cms.gov/browse/MAT-5610)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by RohitKandimalla — APPROVED** [review]

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java [hunk 1]
```diff
@@ -276,15 +277,19 @@ private void buildVsacErrorMessage(CqlCode cqlCode, VsacCode vsacCode) {
...
+        .filter(cse -> isCodeSystemMatch(cse, cqlCodeSystemOid, model))
         .findFirst();
   }
 
+  private boolean isCodeSystemMatch(CodeSystemEntry cse, String oid, String model) {
+    if ("QDM".equals(model)) {
+      return cse.getOid().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
+    }
+    return cse.getUrl().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
+  }
```
[the remark this answers is not in the document]
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java:291]: Perhaps:
```suggestion
    return cse.getOid().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid))
        || cse.getUrl().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
  }
```

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java [hunk 2]
```diff
@@ -276,15 +277,19 @@ private void buildVsacErrorMessage(CqlCode cqlCode, VsacCode vsacCode) {
...
-                    .equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(cqlCodeSystemOid)))
+        .filter(cse -> isCodeSystemMatch(cse, cqlCodeSystemOid, model))
         .findFirst();
   }
 
+  private boolean isCodeSystemMatch(CodeSystemEntry cse, String oid, String model) {
+    if ("QDM".equals(model)) {
+      return cse.getOid().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
+    }
+    return cse.getUrl().equalsIgnoreCase(TerminologyServiceUtil.sanitizeInput(oid));
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/service/VsacService.java:290]: Trying to see if the `if` could be safely removed.

Perhaps a simple `or` will suffice?
```suggestion
    return StringUtils.equalsIgnoreCase(cse.getOid(),TerminologyServiceUtil.sanitizeInput(oid))
        || StringUtils.equalsIgnoreCase(cse.getUrl(), TerminologyServiceUtil.sanitizeInput(oid));
```

Comment (adongare) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java:290]: I think that is a good approach but that way we will be allowing QDM style declaration in FHIR and vice versa, right?

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java:290]: Alas, it would, nice catch! Please ignore my suggestion.
